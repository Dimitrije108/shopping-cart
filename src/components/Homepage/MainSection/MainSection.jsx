import { Link } from "react-router-dom";
import styles from "./MainSection.module.css";

export default function MainSection() {
	return (
		<section className={styles.mainHomepageSection}>
			<div className={styles.heroImage}></div>
			<div className={styles.mainContent}>
				<h1>
					From Naboo to the Outer Rim — Corellian Engineering Excellence Now in Your Hands!
				</h1>
				<Link to="/starships" data-testid="starshipsLink">
					<button className={styles.shopBtn}>
						Explore the Galaxy — Shop Here!
					</button>
				</Link>
			</div>
		</section>
	)
};
