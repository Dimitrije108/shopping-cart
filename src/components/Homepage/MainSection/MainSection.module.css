.mainHomepageSection {
	min-width: 100%;
	min-height: 103.4vh;

	display: flex;
	flex-direction: column;
	justify-content: flex-end;
}

.heroImage {
	width: 100%;
	min-height: 110vh;

	background-image: url("/src/assets/images/phantom-menace-ferguson.jpeg"), 
										linear-gradient(to bottom, rgba(0, 0, 0, 0) 50%, rgba(0, 0, 0, 1) 100%);
	background-size: cover;
	background-repeat: no-repeat;
	background-position: top;
  background-blend-mode: overlay;

	position: absolute;
	top: 0;
	left: 0;
}

.mainContent {
	margin-bottom: 15rem;
	padding: 1rem;
	max-width: 1250px;

	background-color: rgba(0, 0, 0, 0.3);
	align-self: center;
	box-sizing: border-box;

	z-index: 1;
}

.mainContent > h1 {
	font-family: 'DINNext bold';
	font-size: 2.6rem;
}

.shopBtn {
	margin-top: 1rem;
	padding: 1rem 4rem;

	font-family: 'DINNext bold';
	font-size: 1.5rem;
	letter-spacing: 1px;
	
  border: 2px solid rgb(179, 179, 179);

	align-self: flex-end;

	transition: all 0.3s ease-in-out;
}

.shopBtn:hover {
  border: 2px solid rgb(238, 238, 238);
	letter-spacing: 2px;
	background-color: rgba(0, 0, 0, 0.2);
}

@media (max-width: 900px) {
  .mainContent > h1 {
		font-size: 2rem;
	}

	.shopBtn {
		padding: 0.8rem 3rem;
		font-size: 1.3rem;
		letter-spacing: 1px;
	}
}

@media (max-width: 768px) {
	.mainContent {
		margin-bottom: 13rem;
	}

	.mainContent > h1 {
		font-size: 1.7rem;
	}
}

@media (max-width: 660px) {
	.shopBtn {
		letter-spacing: 0.5px;
	}

	.shopBtn:hover {
		letter-spacing: 1px;
	}
}
