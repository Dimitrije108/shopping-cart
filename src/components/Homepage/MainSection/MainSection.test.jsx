import { describe, it, expect } from "vitest";
import { render, screen  } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import MainSection from "./MainSection";

describe("Homepage's Main Section component", () => {
	it("Main Section snapshot", () => {
		const mainSection = render(
			<MemoryRouter>
				<MainSection />
			</MemoryRouter>
		)

		expect(mainSection).toMatchSnapshot();
	});

	it("Main Section elements render correctly", () => {
		render(
			<MemoryRouter>
				<MainSection />
			</MemoryRouter>
		)

		const heading = screen.getByRole("heading", { name: "From Naboo to the Outer Rim — Corellian Engineering Excellence — Now in Your Hands!" });
		const link = screen.getByTestId("starshipsLink");
		const button = screen.getByRole("button", { name: "Explore the Galaxy — Shop Here!" })

		expect(heading).toBeInTheDocument();
		expect(link).toHaveAttribute("href", "/starships");
		expect(button).toBeInTheDocument();
	});
});
