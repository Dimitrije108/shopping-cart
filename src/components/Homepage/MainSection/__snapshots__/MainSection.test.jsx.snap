// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage's Main Section component > Main Section snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="_mainHomepageSection_27eff8"
      >
        <div
          class="_heroImage_27eff8"
        />
        <div
          class="_mainContent_27eff8"
        >
          <h1>
            From Naboo to the Outer Rim — Corellian Engineering Excellence — Now in Your Hands!
          </h1>
          <a
            data-discover="true"
            data-testid="starshipsLink"
            href="/starships"
          >
            <button
              class="_shopBtn_27eff8"
            >
              Explore the Galaxy — Shop Here!
            </button>
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="_mainHomepageSection_27eff8"
    >
      <div
        class="_heroImage_27eff8"
      />
      <div
        class="_mainContent_27eff8"
      >
        <h1>
          From Naboo to the Outer Rim — Corellian Engineering Excellence — Now in Your Hands!
        </h1>
        <a
          data-discover="true"
          data-testid="starshipsLink"
          href="/starships"
        >
          <button
            class="_shopBtn_27eff8"
          >
            Explore the Galaxy — Shop Here!
          </button>
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
